com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/plugin/PluginScanner.java

```diff
@@ -48,24 +48,51 @@ public class PluginScanner {
 	 */
 	public static <T> T getOrCreatePlugin(String packageName, Class<T> pluginBaseClass) throws CompileExceptionError {
 
-		IClassScanner scanner = Project.getClassLoaderScanner();
-		if (scanner == null) {
-			Bob.verbose("PluginScanner has no class loader scanner");
-			return null;
+		List<T> plugins = getOrCreatePlugins(packageName, pluginBaseClass);
+		T plugin = null;
+		if (plugins != null) {
+			if (plugins.size() > 1) {
+				throw new CompileExceptionError("PluginScanner found more than one class implementing " + pluginBaseClass + " in package " + packageName);
+			}
+			// get the plugin (or null if none was found) and cache it
+			plugin = (T)plugins.get(0);
 		}
+		return plugin;
+	}
+	
+	/**
+	 * Get a previously cached instances or find and create instances of classes
+	 * extending a specific base class and located within a specific package. The
+	 * class must:
+	 * 
+	 * - Not be private
+	 * - Not be abstract
+	 * - Extend the base class
+	 * 
+	 * @param packageName 
+	 * @param pluginBaseClass
+	 * @return List with class instances or null if no class was found
+	 */
+	public static <T> List<T> getOrCreatePlugins(String packageName, Class<T> pluginBaseClass) throws CompileExceptionError {
 
 		// check if we've already searched for and cached a plugin for this package path and base class
 		// and if that is the case return the cached instance
 		// note that HashMap accepts null as a value (and key for that matter) which means that we'll
 		// return null if we have searched for and not found a plugin
 		String pluginKey = packageName + pluginBaseClass;
 		if (pluginsCache.containsKey(pluginKey)) {
-			T plugin = (T)pluginsCache.get(pluginKey);
-			if (plugin != null)
+			List<T> plugins = (List<T>)pluginsCache.get(pluginKey);
```

Comment:
Now we collect all the plugins, but in a case when bob can't operate more than one plugin it requests only one if more plugins exist for such cases - exception thrown (see line 55)